### PR TITLE
Test cleanup: typeResolver builder on unions and interfaces, applied directive tests, and more

### PR DESCRIPTION
**Pull request by dondonz**
This PR fixes more tests with deprecated methods.

This is the last wave of Groovy test fixes. Next PR will fix deprecated usages in a few Java test/util files.

### CODE REVIEW

File: src/main/java/graphql/schema/GraphQLInterfaceType.java [hunk 1]
```diff
@@ -363,7 +363,13 @@ public Builder clearFields() {
...
         }
 
-
+        /**
+         * @param typeResolver the type resolver
+         *
+         * @return this builder
+         *
+         * @deprecated use {@link graphql.schema.GraphQLCodeRegistry.Builder#typeResolver(GraphQLInterfaceType, TypeResolver)} instead
+         */
```
Comment (dondonz) [inline src/main/java/graphql/schema/GraphQLInterfaceType.java:372]: Add doc to make the alternative clearer

File: src/main/java/graphql/schema/GraphQLUnionType.java [hunk 2]
```diff
@@ -275,6 +275,13 @@ public Builder extensionDefinitions(List<UnionTypeExtensionDefinition> extension
             return this;
         }
 
+        /**
+         * @param typeResolver the type resolver
+         *
+         * @return this builder
+         *
+         * @deprecated use {@link graphql.schema.GraphQLCodeRegistry.Builder#typeResolver(GraphQLUnionType, TypeResolver)} instead
+         */
```
Comment (dondonz) [inline src/main/java/graphql/schema/GraphQLUnionType.java:284]: Add doc to make the alternative clearer

File: src/test/groovy/graphql/GraphQLTest.groovy [hunk 3]
```diff
@@ -1397,28 +1415,6 @@ many lines''']
         e.message.contains("an illegal value for the argument ")
     }
 
-    def "Applied schema directives arguments are validated for programmatic schemas"() {
```
Comment (dondonz) [inline src/test/groovy/graphql/GraphQLTest.groovy:1400]: Moved to another file

File: src/test/groovy/graphql/execution/instrumentation/LegacyTestingInstrumentation.groovy [hunk 4]
```diff
@@ -39,85 +39,85 @@ class LegacyTestingInstrumentation implements Instrumentation {
 
     @Override
     InstrumentationContext<ExecutionResult> beginExecution(InstrumentationExecutionParameters parameters) {
-        assert parameters.getInstrumentationState() == instrumentationState
+        assert parameters.getInstrumentationState() == instrumentationState // Retain for test coverage
```
Comment (dondonz) [inline src/test/groovy/graphql/execution/instrumentation/LegacyTestingInstrumentation.groovy:42]: Everything in this file is intentionally deprecated

File: src/test/groovy/graphql/introspection/IntrospectionTest.groovy [hunk 5]
```diff
@@ -421,7 +421,7 @@ class IntrospectionTest extends Specification {
                                                         .name("inputField")
                                                         .type(GraphQLString))
                                                 .build())
-                                        .defaultValue(new FooBar())
+                                        .defaultValue(new FooBar()) // Retain for test coverage. There is no alternative method that sets an internal value.
```
Comment (dondonz) [inline src/test/groovy/graphql/introspection/IntrospectionTest.groovy:424]: The alternatives are `defaultValueLiteral(Value)` or `defaultValueProgrammatic(Object)`, neither of which work in this case.

File: src/test/groovy/graphql/schema/SchemaTraverserTest.groovy [hunk 6]
```diff
@@ -388,15 +382,6 @@ class SchemaTraverserTest extends Specification {
 
     }
 
-
```
Comment (dondonz) [inline src/test/groovy/graphql/schema/SchemaTraverserTest.groovy:391]: These are old tests, a `typeResolver` is no longer required on the union/interface type

File: src/test/groovy/graphql/schema/idl/SchemaGeneratorTest.groovy [hunk 7]
```diff
@@ -1634,13 +1633,14 @@ class SchemaGeneratorTest extends Specification {
         argInt.getDirective("thirdDirective") != null
 
         def intDirective = argInt.getDirective("intDirective")
-        intDirective.name == "intDirective"
-        intDirective.arguments.size() == 1
-        def directiveArg = intDirective.getArgument("inception")
+        def intAppliedDirective = argInt.getAppliedDirective("intDirective")
```
Comment (dondonz) [inline src/test/groovy/graphql/schema/idl/SchemaGeneratorTest.groovy:1636]: Key change: the applied directive contains the value, and the directive contains the default value

File: src/test/groovy/graphql/schema/validation/TypesImplementInterfacesTest.groovy [hunk 8]
```diff
@@ -20,13 +18,6 @@ import static graphql.schema.GraphQLUnionType.newUnionType
 
 class TypesImplementInterfacesTest extends Specification {
 
-    TypeResolver typeResolver = new TypeResolver() {
```
Comment (dondonz) [inline src/test/groovy/graphql/schema/validation/TypesImplementInterfacesTest.groovy:23]: These are older tests, `typeResolver` not required directly on interface/union type

File: src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationTest.groovy [hunk 9]
```diff
@@ -39,7 +39,6 @@ class DataLoaderDispatcherInstrumentationTest extends Specification {
         }
     }
 
-
```
Comment (bbakerman) [inline src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationTest.groovy:42]: I halways hated that new line

File: src/test/groovy/graphql/schema/validation/TypeAndFieldRuleTest.groovy [hunk 10]
```diff
@@ -10,7 +11,6 @@ import static graphql.schema.GraphQLUnionType.newUnionType
 
 class TypeAndFieldRuleTest extends Specification {
 
-
```
Comment (bbakerman) [inline src/test/groovy/graphql/schema/validation/TypeAndFieldRuleTest.groovy:13]: that new line has bugged me forever!

File: src/test/groovy/graphql/schema/validation/TypeAndFieldRuleTest.groovy [hunk 11]
```diff
@@ -55,7 +55,6 @@ class TypeAndFieldRuleTest extends Specification {
         e.message == "invalid schema:\n\"InputType\" must define one or more fields."
     }
 
-
```
Comment (bbakerman) [inline src/test/groovy/graphql/schema/validation/TypeAndFieldRuleTest.groovy:58]: same with that one

File: src/test/groovy/graphql/schema/GraphQLArgumentTest.groovy [hunk 12]
```diff
@@ -189,4 +195,26 @@ class GraphQLArgumentTest extends Specification {
...
+                newObject()
+                        .name("Query")
+                        .field(field)
+                        .build())
+                .build()
+        then:
+        def e = thrown(InvalidSchemaException)
+        e.message.contains("Invalid argument 'arg' for applied directive of name 'cached'")
+    }
+
```
Comment (bbakerman) [inline src/test/groovy/graphql/schema/GraphQLArgumentTest.groovy:219]: Nice one

File: src/test/groovy/graphql/schema/GraphQLInterfaceTypeTest.groovy [hunk 13]
```diff
@@ -107,4 +105,19 @@ class GraphQLInterfaceTypeTest extends Specification {
...
+                .fields([
+                        newFieldDefinition().name("NAME").type(GraphQLString).build(),
+                        newFieldDefinition().name("NAME").type(GraphQLInt).build()
+                ])
+                .typeResolver(new TypeResolverProxy()) // Retain for test coverage
+                .build()
+        then:
+        interfaceType.getName() == "TestInterfaceType"
+        interfaceType.getFieldDefinition("NAME").getType() == GraphQLInt
+    }
```
Comment (bbakerman) [inline src/test/groovy/graphql/schema/GraphQLInterfaceTypeTest.groovy:122]: nice

**Review by bbakerman — APPROVED** [review]
Great detective work.  Getting rid of deprecation is harder than it looks

**Comment by bbakerman** [comment]
Dont forget to set  a target milestone on PRs

File: src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationTest.groovy (hunk 9 above)
Comment (dondonz) [reply src/test/groovy/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentationTest.groovy:42]: Unnecessary new lines drive me

nuts

**Pull request merged**